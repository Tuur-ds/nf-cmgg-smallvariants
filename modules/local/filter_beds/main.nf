process FILTER_BEDS {
    tag "$meta.id"
    label 'process_single'

    conda "bioconda::bedtools=2.31.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bedtools:2.31.0--hf5e1c6e_2' :
        'biocontainers/bedtools:2.31.0--hf5e1c6e_2' }"

    input:
    tuple val(meta), path(bed)

    output:
    tuple val(meta), path('*.bed'), emit: bed
    path  "versions.yml"          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // Remove regions with no coverage from the callable regions BED file
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    def unzip = bed.extension == "gz" ? "zcat" : "cat"
    """
    ${unzip} ${bed} | grep ${args} | bedtools merge ${args2} > ${prefix}.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bedtools: \$(bedtools --version | sed -e "s/bedtools v//g")
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bedtools: \$(bedtools --version | sed -e "s/bedtools v//g")
    END_VERSIONS
    """
}
