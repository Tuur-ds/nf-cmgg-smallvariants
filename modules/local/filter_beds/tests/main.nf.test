nextflow_process {

    name "Test Process FILTER_BEDS"
    script "../main.nf"
    process "FILTER_BEDS"

    tag "modules"
    tag "modules_local"
    tag "filter_beds"

    test("homo_sapiens - bed") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.bed, checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - bed - stub") {

        options "-stub"
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.bed, checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}