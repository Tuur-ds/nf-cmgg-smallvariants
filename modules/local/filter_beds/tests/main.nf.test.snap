{
    "homo_sapiens - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,fa245abf8add7a80650566a1de67ec04"
                    ]
                ],
                "1": [
                    "versions.yml:md5,38ea76fdc8d681b1e47415b195cccd88"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,fa245abf8add7a80650566a1de67ec04"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,38ea76fdc8d681b1e47415b195cccd88"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T17:12:06.470648263"
    },
    "homo_sapiens - bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,38ea76fdc8d681b1e47415b195cccd88"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,38ea76fdc8d681b1e47415b195cccd88"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T17:09:43.531020092"
    }
}